apm-sniffer/apm-sdk-plugin/hbase-1.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/hbase/HTableInterceptor.java

```diff
@@ -0,0 +1,129 @@
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.plugin.hbase;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hbase.client.ClusterConnection;
+import org.apache.hadoop.hbase.client.HTable;
+import org.apache.hadoop.hbase.client.OperationWithAttributes;
+import org.apache.skywalking.apm.agent.core.context.CarrierItem;
+import org.apache.skywalking.apm.agent.core.context.ContextCarrier;
+import org.apache.skywalking.apm.agent.core.context.ContextManager;
+import org.apache.skywalking.apm.agent.core.context.tag.Tags;
+import org.apache.skywalking.apm.agent.core.context.trace.AbstractSpan;
+import org.apache.skywalking.apm.agent.core.context.trace.SpanLayer;
+import org.apache.skywalking.apm.agent.core.logging.api.ILog;
+import org.apache.skywalking.apm.agent.core.logging.api.LogManager;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.EnhancedInstance;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.InstanceConstructorInterceptor;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.InstanceMethodsAroundInterceptor;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.MethodInterceptResult;
+import org.apache.skywalking.apm.agent.core.util.CollectionUtil;
+import org.apache.skywalking.apm.network.trace.component.ComponentsDefine;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+public class HTableInterceptor implements InstanceMethodsAroundInterceptor, InstanceConstructorInterceptor {
+
+    private static final ILog logger = LogManager.getLogger(HTableInterceptor.class);
+    private static final String PREFIX_OPERATION_NAME = "/Htable/";
+    private static final String HBASE_DB_TYPE = "hbase";
+
+    @Override
+    @SuppressWarnings("unchecked, rawtypes")
+    public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allArguments, Class<?>[] argumentsTypes,
+                             MethodInterceptResult result) throws Throwable {
+        boolean canTracingServer = false;
+        List<OperationWithAttributes> operations = null;
+        OperationWithAttributes operation = null;
+        if (allArguments != null && allArguments.length > 0) {
+            if (allArguments[0] instanceof List) {
+                List list = (List) allArguments[0];
+                if (!CollectionUtil.isEmpty(list) && list.get(0) instanceof OperationWithAttributes) {
+                    operations = list;
+                    canTracingServer = true;
+                }
+            } else if (allArguments[0] instanceof OperationWithAttributes) {
+                operation = (OperationWithAttributes) allArguments[0];
+                canTracingServer = true;
+            }
+        }
+        AbstractSpan span;
+        if (canTracingServer) {
```

Comment:
What is the case `not instanceof OperationWithAttributes`?

Reply:
I don't know how all versions of Hbase-Client evolve, just in case

Reply:
OK, got it:)